sklearn/model_selection/_search.py

```diff
@@ -1044,7 +1042,10 @@ class GridSearchCV(BaseSearchCV):
     def __init__(self, estimator, param_grid, scoring=None, fit_params=None,
                  n_jobs=1, iid=True, refit=True, cv=None, verbose=0,
                  pre_dispatch='2*n_jobs', error_score='raise',
-                 return_train_score=True):
+                 return_train_score=False):
+        warnings.warn(
```

Comment:
Do this in fit, please.

The same change needs to be made for RandomizedSearchCV, or needs to be done in BaseSearchCV